src/browser/FileProxy.js

```diff
@@ -141,103 +147,121 @@
+                        }
+                    } else {
+                        // Otherwise, if no other error occurs, getFile must return a FileEntry
+                        // corresponding to path.
 
-                    successCallback(fileEntryFromIdbEntry(fileEntry));
-                }
-            }, errorCallback);
+                        successCallback(fileEntryFromIdbEntry(fileEntry));
+                    }
+                },
+                errorCallback
+            );
         };
 
         exports.getFileMetadata = function (successCallback, errorCallback, args) {
             var fullPath = args[0];
 
-            exports.getFile(function (fileEntry) {
-                successCallback(new File(fileEntry.file_.name, fileEntry.fullPath, '', fileEntry.file_.lastModifiedDate,
-                    fileEntry.file_.size));
-            }, errorCallback, [fullPath, null]);
+            exports.getFile(
+                function (fileEntry) {
+                    successCallback(
+                        new File(fileEntry.file_.name, fileEntry.fullPath, '', fileEntry.file_.lastModifiedDate, fileEntry.file_.size)
+                    );
+                },
+                errorCallback,
+                [fullPath, null]
+            );
         };
 
         exports.getMetadata = function (successCallback, errorCallback, args) {
-            exports.getFile(function (fileEntry) {
-                successCallback(
-                    {
+            exports.getFile(
+                function (fileEntry) {
+                    successCallback({
                         modificationTime: fileEntry.file_.lastModifiedDate,
                         size: fileEntry.file_.lastModifiedDate
                     });
-            }, errorCallback, args);
+                },
+                errorCallback,
+                args
+            );
         };
 
         exports.setMetadata = function (successCallback, errorCallback, args) {
             var fullPath = args[0];
             var metadataObject = args[1];
 
-            exports.getFile(function (fileEntry) {
-                fileEntry.file_.lastModifiedDate = metadataObject.modificationTime;
-                idb_.put(fileEntry, fileEntry.file_.storagePath, successCallback, errorCallback);
-            }, errorCallback, [fullPath, null]);
+            exports.getFile(
+                function (fileEntry) {
+                    fileEntry.file_.lastModifiedDate = metadataObject.modificationTime;
+                    idb_.put(fileEntry, fileEntry.file_.storagePath, successCallback, errorCallback);
+                },
+                errorCallback,
+                [fullPath, null]
+            );
         };
 
         exports.write = function (successCallback, errorCallback, args) {
             var fileName = args[0];
             var data = args[1];
             var position = args[2];
-            var isBinary = args[3]; // eslint-disable-line no-unused-vars
```

Comment:
No reason to allocate memory if the variable is not used.

Reply:
remove it.